Zend/zend_alloc.c

```diff
@@ -1065,6 +1065,9 @@ static void *zend_mm_alloc_pages(zend_mm_heap *heap, uint32_t pages_count ZEND_F
 
 found:
 	if (steps > 2 && pages_count < 8) {
+		ZEND_MM_CHECK(chunk->next->prev == chunk, "zend_mm_heap corrupted");
```

Comment:
wouldn't `chunk->next->prev == chunk && chunk->prev->next == chunk` be faster?

Reply:
in clang it is faster: https://godbolt.org/z/5767x31GY

Reply:
> in clang it is faster: https://godbolt.org/z/5767x31GY

It's almost as if contrived examples don't represent real-world workload:

https://godbolt.org/z/8xaM1bYn5